src/Strategy/VSI.hs

```diff
@@ -80,10 +80,12 @@ transformLocator :: ValidVSILocator -> Dependency
 transformLocator locator =
   Dependency (validType locator) (validName locator) (CEq <$> validRevision locator) [] [] mempty
 
-analyze :: (Has (Lift IO) sig m, MonadIO m, Has Exec sig m, Has Diagnostics sig m) => WigginsOpts -> m (Graphing Dependency)
+-- TODO: confirm this is Partial @kit
+analyze :: (Has (Lift IO) sig m, MonadIO m, Has Exec sig m, Has Diagnostics sig m) => WigginsOpts -> m (Graphing Dependency, GraphBreadth)
 analyze opts = context "VSI" $ do
   vsiLocators <- context "Running VSI binary" $ withWigginsBinary (execWigginsJson opts)
-  context "Building dependency graph" $ fromEither (buildGraph vsiLocators)
+  graph <- context "Building dependency graph" $ fromEither (buildGraph vsiLocators)
+  pure (graph, Partial)
```

Comment:
cc @kitified

We're trying to determine if this strategy should be applied recursively.  `Partial` strategies are recursive, `Complete` strategies are non-recursive.

Reply:
The idea behind the VSI strategy is that it discovers a single target which is the project scan root. It then uploads all of the fingerprints to the VSI service, and reports a single flat dep graph of the direct dependencies found by the VSI service for the singular target, which encompasses the entire scan directory.

So if I'm understanding the change set properly, this strategy should be considered `Complete`.

Reply:
thank you @kitified